pkg/nfd-master/nfd-master.go

```diff
@@ -475,11 +474,8 @@ func (m *nfdMaster) Stop() {
 // Wait until NfdMaster is able able to accept connections.
 func (m *nfdMaster) WaitForReady(timeout time.Duration) bool {
 	select {
-	case ready, ok := <-m.ready:
-		// Ready if the flag is true or the channel has been closed
-		if ready || !ok {
-			return true
-		}
+	case <-m.ready:
+		return true
```

Comment:
I think we need to re write the entire func, the last return will never be used

Reply:
> I think we need to re write the entire func, the last return will never be used

True, I modified the func to eliminate unreachable parts

Reply:
pardon my ignorance, should this `select` be inside an empty `for` ?

Reply:
No, the select blocks until the channel is closed or the timer expoires